#153114592 Add daily, weekly and monthly periodic configurations

fnshemerirwe opened the pull request:
#### What does this PR do?
Enable a user to configure after how long they want to be receiving the report, whether daily, weekly or monthly.

#### Description of the tasks to be completed?
1. Have a user select from the UI if they would like to receive reports and after how long they want to be receiving these reports in `profile.html`.
2. Add a new table column to store the report duration in the `Profile` model in `accounts.models.py`.
3. Update the `next_report_date` whenever a report is sent in he `Profile` model in `accounts.models.py` or when a new report duration is selected in `profile.html`.
4. Add triggers for daily, weekly and monthly reports in `sendreports.py`.
5. Write tests for the daily, weekly and monthly triggers in `sendreports.py`.

#### How should this be manually tested.
After cloning the repo, cd into the root of the project and run `./manage.py test hc.api.tests.test_sendreports`.

#### Any background context you want to provide?
 * The application is hosted on heroku at https://healthchck.herokuapp.com/.
 * Travis tests are still failing for pep8 as HEALTHCHECKS is still in development.

#### What are the relevant pivotal tracker stories?
[#153114592](https://www.pivotaltracker.com/n/projects/2131074/stories/153114592)

#### Screenshots
<img width="941" alt="screen shot 2017-12-12 at 4 55 25 pm" src="https://user-images.githubusercontent.com/9306725/33888215-766d3890-df5d-11e7-8d1d-2f8f9e6be49d.png">

fnshemerirwe committed:
[Feature #153114592] Add radio buttons for required periods.

fnshemerirwe committed:
[Feature #153114592] Add radio buttons for required periods.

silverjimmy committed:
[Feature

silverjimmy committed:
[Feature

silverjimmy committed:
[Feature

silverjimmy committed:
[Feature

silverjimmy committed:
[Feature

silverjimmy committed:
[Feature

silverjimmy committed:
[Chore#153114590] Removed Implemented JS.

silverjimmy committed:
[Chore#153114590] Removed Implemented JS.

fnshemerirwe committed:
[Feature #153114592] Add report duration to models

fnshemerirwe committed:
[Feature #153114592] Add report duration to models

fnshemerirwe committed:
[Feature #153114592] Add and handle periodic report times

fnshemerirwe committed:
[Feature #153114592] Add and handle periodic report times

fnshemerirwe committed:
[Feature #153114592] Handle send reports

fnshemerirwe committed:
[Feature #153114592] Handle send reports

fnshemerirwe committed:
[Feature #153114592] Solve merge conflict

silverjimmy committed:
[Feature #153114592] Fix margin in dashboard

silverjimmy committed:
[Feature #153114592] Fix margin in dashboard

fnshemerirwe committed:
Merge branch 'develop' of https://github.com/andela/hc-gnext into ft-periodic-reports-configuration-153114592

Make branch ft-periodic-reports-configuration-153114592 up-to-date with branch develop.

fnshemerirwe committed:
[Feature #153114592] Sync the settings in the UI with settings saved in the database.

fnshemerirwe committed:
[Feature #153114592] Sync the settings in the UI with settings saved in the database.

silverjimmy committed:
[Feature #153114592] Report Dashboard link added

silverjimmy committed:
[Feature #153114592] Report Dashboard link added

silverjimmy committed:
[Feature #153114592] Model to handle requests to the Report page

silverjimmy committed:
[Feature #153114592] Model to handle requests to the Report page

silverjimmy committed:
[Feature #153114592] Missing column name added

silverjimmy committed:
[Feature #153114592] Missing column name added

silverjimmy committed:
[Feature #153114592] Model to handle requests to the Report page

silverjimmy committed:
[Feature #153114592] Template for the report dashboard

silverjimmy committed:
[Feature #153114592] Template for the report dashboard

silverjimmy committed:
[Feature #153114592] Template for the report dashboard

silverjimmy committed:
[Feature #153114592] Template for the report dashboard

silverjimmy committed:
[Feature #153114592] handle requests to the Report page

silverjimmy committed:
[Feature #153114592] handle requests to the Report page

silverjimmy committed:
[Feature #153114592] Added Reports to the dashboard

silverjimmy committed:
[Feature #153114592] Added Reports to the dashboard

silverjimmy committed:
[Feature #153114592] Switched report from using seconds to days

silverjimmy committed:
[Feature #153114592] Switched report from using seconds to days

silverjimmy committed:
[Feature #153114592] Procfile to be used by heroku on how to run the app

silverjimmy committed:
[Feature #153114592] Tells heroku which python version to run

silverjimmy committed:
[Feature #153114592] Added in module to help with running app on heroku

silverjimmy committed:
[Feature #153114592] Added in module to help with running app on heroku

silverjimmy committed:
Committing before pushing to Heroku.

silverjimmy committed:
Committing before pushing to Heroku.

silverjimmy committed:
Committing before pushing to Heroku.

silverjimmy committed:
Committing before pushing to Heroku.

silverjimmy committed:
Added production settings file.

silverjimmy committed:
Added production settings file.

silverjimmy committed:
Added production settings file.

silverjimmy committed:
Added production settings file.

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

silverjimmy committed:
Initial commit of work

fnshemerirwe committed:
[Feature #153114592] Move SCHEDULE_CHOICES to the beginning of the file

fnshemerirwe committed:
[Feature #153114592] Move SCHEDULE_CHOICES to the beginning of the file

fnshemerirwe committed:
[Feature #153114592] Remove pep8 style issues

fnshemerirwe committed:
[Feature #153114592] Remove pep8 style issues

fnshemerirwe committed:
[Feature #153114592] Add tests for send report

fnshemerirwe committed:
[Feature #153114592] Add tests for send report

fnshemerirwe committed:
Merge branch 'ft-periodic-reports-configuration-153114592' of https://github.com/andela/hc-gnext into ft-periodic-reports-configuration-153114592

fnshemerirwe committed:
[Feature #153114592] Add freezegun to requirements.txt

fnshemerirwe committed:
[Feature #153114592] Add freezegun to requirements.txt

silverjimmy committed:
testing

silverjimmy committed:
testing

fnshemerirwe committed:
[Feature #153114592] Fix pep8 style issues

fnshemerirwe committed:
[Feature #153114592] Fix pep8 style issues

fnshemerirwe committed:
[Feature #153114592] Fix pep8 style issues

fnshemerirwe committed:
[Feature #153114592] Fix pep8 style issues

silverjimmy committed:
changed Site Root url to point to new server

isheebo commented:
- Change the Pivotal tracker story to a URL. Use the story URL instead of the story ID
- The Heroku deployment is a bit faulty. When i click the Login link sent to my email, it brings up an error as shown in this screenshot below.
<img width="922" alt="screen shot 2017-12-12 at 6 57 10 pm" src="https://user-images.githubusercontent.com/29917291/33894189-65f6d438-df6e-11e7-8998-16960f3b3117.png">

- I would recommend you add the link to your Heroku application (https://healthchck.herokuapp.com) under the **`How can this be manually tested?`** section. I believe this would make it easier for a non-technical person to see and actually test the highlighted functionality of the code.

silverjimmy committed:
changed Site Root url to point to new server

isheebo reviewed (commented):
- Change the Pivotal tracker story to a URL. Use the story URL instead of the story ID
- The Heroku deployment is a bit faulty. When i click the Login link sent to my email, it brings up an error as shown in this screenshot below.
<img width="922" alt="screen shot 2017-12-12 at 6 57 10 pm" src="https://user-images.githubusercontent.com/29917291/33894189-65f6d438-df6e-11e7-8998-16960f3b3117.png">

- I would recommend you add the link to your Heroku application (https://healthchck.herokuapp.com) under the **`How can this be manually tested?`** section. I believe this would make it easier for a non-technical person to see and actually test the highlighted functionality of the code.

silverjimmy committed:
Merge branch 'ft-periodic-reports-configuration-153114592' of https://github.com/andela/hc-gnext into ft-periodic-reports-configuration-153114592

fnshemerirwe committed:
[Feature #153114592] Add date to subject to avoid grouped emails

fnshemerirwe committed:
[Feature #153114592] Add date to subject to avoid grouped emails

fnshemerirwe committed:
[Feature #153114592] Remove unnecessary dependencies

fnshemerirwe committed:
[Feature #153114592] Remove unnecessary dependencies

TinaMurimi commented on hc/api/management/commands/sendreports.py:
```diff
@@ -27,7 +27,7 @@ def add_arguments(self, parser):
         )
 
-    def handle_one_run(self):
+    def handle_month_run(self):
```
`handle_month_run`, `handle_week_run`, `and handle_daily_run` could be one method that has arguments for the number of days `1`, `7`, or `30`

TinaMurimi reviewed (commented):
Not using Whitenoise?

fnshemerirwe committed:
[Feature #153114592] Combine handle periodic reports within one method

fnshemerirwe committed:
[Feature #153114592] Combine handle periodic reports within one method

silverjimmy committed:
Merge branch 'ft-periodic-reports-configuration-153114592' of https://github.com/andela/hc-gnext into ft-periodic-reports-configuration-153114592

silverjimmy committed:
[feature #153114592] whitenoise added to handle static files

silverjimmy committed:
[feature #153114592] whitenoise added to handle static files

silverjimmy committed:
[feature #153114592] Static files created by whitenoise

silverjimmy committed:
[feature #153114592] Static files created by whitenoise

TinaMurimi commented on hc-venv/.Python:
```diff
@@ -0,0 +1 @@
+/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/Python
```
This file should not be in the repo

silverjimmy replied:
Issue has been resolved, there was an issue with the naming of my env in the gitignore file

TinaMurimi commented on hc-venv/bin/activate:
```diff
@@ -0,0 +1,78 @@
+# This file must be used with "source bin/activate" *from bash*
```
Why are your env files in the repo?

silverjimmy replied:
Issue has been resolved, there was an issue with the naming of my env in the gitignore file

TinaMurimi reviewed (commented):
5028 files changed?

<img width="905" alt="screen shot 2017-12-14 at 10 51 36" src="https://user-images.githubusercontent.com/26300276/33980756-4a35e0ec-e0bb-11e7-9917-b5144bfd3d55.png">

silverjimmy committed:
[feature #153114592] Removed the hc-venv file

silverjimmy committed:
[feature #153114592] Removed the hc-venv file

silverjimmy committed:
[feature #153114592] Removed the hc-venv file

silverjimmy committed:
Merge branch 'develop' into ft-periodic-reports-configuration-153114592

giddke committed:
[Feature #153114597]Add sms icon

giddke committed:
[Feature #153114597]Add SMS integration

giddke committed:
[Feature #153114597]Add sms to channel kinds choices

giddke committed:
[Feature #153114597]Add sms icon

silverjimmy committed:
[Feature #153114597] Added sms url to channel_urls

silverjimmy committed:
[Feature #153114597] Add sms url to channel_urls

silverjimmy committed:
[Feature #153114597] Created method to handle sms

silverjimmy committed:
[Feature #153114597] Create method to handle sms

silverjimmy committed:
[Feature #153114597] UI for the sms integration

silverjimmy committed:
[Feature #153114597] UI for the sms integration

silverjimmy committed:
[Feature #153114597] Added in the sms UI

silverjimmy committed:
[Feature #153114597] Add in the sms UI

silverjimmy committed:
[Feature #153114597] Add SMS to channels summary dashboard

silverjimmy committed:
Merge branch 'ft-send-alerts-153114597' of https://github.com/andela/hc-gnext into ft-send-alerts-153114597

silverjimmy committed:
[Feature #153114597] Add AfricasTalking UI for integration

silverjimmy committed:
[Feature #153114597] Add AfricasTalking UI for integration

silverjimmy committed:
[Feature #153114597] Rename sms prefix to africastalking

silverjimmy committed:
[Feature #153114597] Rename sms prefix to africastalking

giddke committed:
[Feature #153114597]Add username and api_key fields

giddke committed:
[Feature #153114597]Add africastalking form

isheebo committed:
[Feature #153114593] Add the interval slider to the timeout-modal

The interval slider allows a user to select a time interval after which periodic updates and
alerts will be sent to their email once a check goes down.

giddke committed:
[Feature #153114597]Add africastalking form

giddke committed:
[Feature #153114597]Reformat if statements

giddke committed:
[Feature #153114597]Add africastalking template

giddke committed:
[Feature #153114597]Add new migrations files for the newly added model fields

Commit:
[Feature #153114596] Add new status choice for fast jobs

Commit:
[Feature #153114596] Modify badge functionality to return status fast

Commit:
[Feature #153114596] Add color for the fast status

Commit:
[Feature #153114596] Add badge for fast jobs

Commit:
[Feature #153114596] Add check for status fast

silverjimmy committed:
Merge branch 'develop' into ft-periodic-reports-configuration-153114592

jonathankamau reviewed (changes requested):
You seem to have alot of unnecessary files in this PR. Please check on what isn't necessary then add it to `.gitignore`

fnshemerirwe committed:
[Feature #153114592] Resolve merge conflict.

giddke committed:
[Feature #153114597]Add migarations file

giddke committed:
[Feature #153114597]Add call to africastalking transport

giddke committed:
[Feature #153114597]Add implementation for africastalking transport class

giddke committed:
[Feature #153114597]Add africastalking to requirements

giddke committed:
[Feature #153114597]Add send sms feature

fnshemerirwe committed:
[Feature #153114596] Add check for reverse grace

fnshemerirwe committed:
[Feature #153114596] Add badges for early and fast statuses

silverjimmy committed:
[Feature #153114597] Add images to show how to intergrate africastalkingsms

silverjimmy committed:
[Feature #153114597] Improve UI for easy understanding of the service

isheebo committed:
[Feature #153114593] Modify files to add the nag interval

isheebo committed:
[Feature #153114593] Add the interval field to the database

isheebo committed:
[Feature #153114593] connect the nag interval slider to the database

isheebo committed:
[Feature #153114593] Add migration files

geofrocker committed:
[Chore #153114593] Add waitress to serve the application on heroku

geofrocker committed:
[Chore #153114593] Add python as the runtime environment

geofrocker committed:
[Chore #153114593] Add whitenoise to serve static files

geofrocker committed:
[Chore #153114593] Add nag functionality for the failed checks

giddke committed:
[Feature #153114597]Remove unnecessary print statements and fix text message formatting

geofrocker committed:
[Chore #153114593] Add heroku requirements

geofrocker committed:
[Chore #153114593] Configure email settings and dj-database

geofrocker committed:
[Chore #153114593] Configure columns for the nag feature

geofrocker committed:
[Chore #153114593] Configure columns interval for nag in the check properties

geofrocker committed:
[Chore #153114593] Remove vscode requirements

geofrocker committed:
[Chore #153114593] Add migrations for the new nag columns

geofrocker committed:
[Chore #153114593] Add whitenoise to serve static files

Commit:
Add color to the newly created glyphicon

Commit:
[Feature #153114596] Modify bootstrap to cater for the glyphicon

Commit:
[Feature #153114596]Modify color for glyphicon

Commit:
[Feature #153114596] Add badge for status fast

Commit:
[Feature #153114596] Add glyphicon to reflect fast status

Commit:
Merge branch 'ft-run-too-often-notification-153114596' of https://github.com/andela/hc-gnext into ft-run-too-often-notification-153114596

fnshemerirwe committed:
[Feature #153114596] Add labels for early and fast statuses

fnshemerirwe committed:
[Feature #153114596] Remove unnecessary changes to files

Commit:
[Feature #153114596] remove the env files

Commit:
[Feature #153114596] Add icon for reverse grace.

silverjimmy committed:
[Feature #153114597] Add new line to show blocks

silverjimmy committed:
Merge branch 'ft-send-alerts-153114597' of https://github.com/andela/hc-gnext into ft-send-alerts-153114597

isheebo committed:
[Feature #153114593] Resolve Merge Conflicts

silverjimmy committed:
make it better

Commit:
[Feature #153114596] Modify the url for heroku deployment

silverjimmy committed:
[Feature #153114597] Update africastalking.html with new line

geofrocker committed:
[Chore #153114593] Add a last blank line

silverjimmy committed:
Merge branch 'develop' into ft-send-alerts-153114597

Commit:
Merge pull request #9 from andela/ft-send-alerts-153114597

#153114597 Send alerts via SMS

isheebo committed:
[Feature #153114593] Edit project settings

isheebo committed:
[Feature #153114593] Resolve Merge Conflicts

isheebo committed:
Merge branch 'develop' into ft-nagging_user-153114593

ivanatu committed:
Merge branch 'develop' into ft-run-too-often-notification-153114596

Commit:
Merge pull request #10 from andela/ft-nagging_user-153114593

#153114593 Send alerts to the user when a job/check has failed until they respond

ivanatu committed:
Merge branch 'develop' into ft-run-too-often-notification-153114596

Commit:
Merge pull request #11 from andela/ft-run-too-often-notification-153114596

#153114596 Notification for jobs running too fast

silverjimmy committed:
[feature #153114592] Remove staticfiles created by whitenoise

silverjimmy committed:
[feature #153114592] Remove Static files created by whitenoise

jonathankamau commented on hc/api/admin.py:
```diff
@@ -162,6 +162,8 @@ def formatted_kind(self, obj):
             return "Pushbullet"
         elif obj.kind == "po":
             return "Pushover"
+        elif obj.kind == "aft":
```
what does `aft` signify? could it be more descriptive

silverjimmy replied:
`aft` isnt a work on this branch it was was a job done in send alerts done by I and Gideon, gave it that abbreviation because looking at the way the rest of the files have been named like `po`, `aft` means 'AfricasTalking`

jonathankamau replied:
So remember what I said on the channel, for any new variable, key etc that you add ensure it is descriptive enough. That should be the guiding standard not what is already on the codebase.

jonathankamau commented on hc/settings.py:
```diff
@@ -148,7 +144,9 @@
 EMAIL_HOST = 'smtp.gmail.com'
 EMAIL_PORT = 587
 EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER')  # your email
+EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER')  # your email
```
isn't this a duplicate of what is in line 146?

silverjimmy replied:
Working on it

jonathankamau commented on requirements.txt:
```diff
@@ -1,16 +1,15 @@
 Django==1.10
 django_compressor==2.1
 djmail==0.11.0
+freezegun==0.3.9
```
what does `freezegun` do?

silverjimmy replied:
That came around when Flavia was writing tests for the assigned task.

jonathankamau replied:
That doesn't quite answer the question. What exactly does it do? What is it's role in the app?

silverjimmy replied:
it's being used in the test file `hc/api/tests/test_sendreports.py` to  mock the default 
 datetime module using `freeze_time`, needed it for our test file to run and to be sure.

jonathankamau commented on templates/accounts/reports_desktop.html:
```diff
@@ -0,0 +1,100 @@
+        </td>
+    </tr>
+    {% endfor %}
+</table>
```
a new line is required here.

silverjimmy replied:
working on it

jonathankamau commented on templates/emails/report-subject.html:
```diff
@@ -1,2 +1,4 @@
+{% if report_duration == 1 %} Daily
+{% elif report_duration == 7%} Weekly
+{% elif report_duration == 30 %} Monthly
+{% endif %} Report as on {{ now }}
```
A new line is required here as well.

silverjimmy replied:
I have sorted that bit

jonathankamau commented on templates/front/my_checks_desktop.html:
```diff
@@ -116,3 +125,5 @@
 
 </table>
+
+
```
why two blank lines? remove one of them.

silverjimmy replied:
I have sorted this issue too.

silverjimmy committed:
[feature #153114592] Correct files having pylint and pep8 issues.

jonathankamau commented on templates/accounts/reports.html:
```diff
@@ -0,0 +1,49 @@
+<script src="{% static 'js/clipboard.min.js' %}"></script>
+<script src="{% static 'js/checks.js' %}"></script>
+{% endcompress %}
+{% endblock %}
```
a new line is needed here.

silverjimmy replied:
I have added a new line to the file

silverjimmy committed:
[feature #153114592] Insert line to meet pylint

silverjimmy committed:
[feature #153114592] Remove unnecessary changes to settings.py
